packages/dev/core/src/FlowGraph/flowGraphPath.ts

```diff
@@ -0,0 +1,82 @@
+import type { FlowGraphContext } from "./flowGraphContext";
+
+const SEPARATORS = /\/|\./;
+/*
+ * This class represents a path of type /x/{y}/z/.../w that is evaluated
+ * on a target object. The string between curly braces ({y} in the example)
+ * is a special template string that is replaced during runtime.
+ */
+export class FlowGraphPath {
+    private path: string;
+    private templateSubstitutions: {
+        [key: string]: number;
+    } = {}; // this is a map of template strings to values that are substituted during runtime
+    public hasTemplateStrings: boolean = false;
+
+    constructor(path: string) {
+        this.path = path;
+        const templateStrings = this._getTemplateStringsInPath(path);
+        this.hasTemplateStrings = templateStrings.length > 0;
+        for (const templateString of templateStrings) {
+            // Part of the path enclosed in curly braces = template string, we'll set a default
+            // value on the template substitutions for now
+            this.templateSubstitutions[templateString] = -1;

```

Comment:
It may be a bit dangerous to use -1 as the invalid value, in case the spec is changed in the future to allow negative indices (as in Python, where it is an offset added to `array.length`)...

Reply:
Good point, I'll use NaN instead